src/main/components/canvas/mouse-eventlistener.tsx

```diff
@@ -0,0 +1,273 @@
+import React, { Component, ComponentType } from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import { ApollonMode } from '../../services/editor/editor-types';
+import { UMLElementRepository } from '../../services/uml-element/uml-element-repository';
+import { AsyncDispatch } from '../../utils/actions/actions';
+import { ModelState } from '../store/model-state';
+import { CanvasContext } from './canvas-context';
+import { withCanvas } from './with-canvas';
+import { UMLElementState } from '../../services/uml-element/uml-element-types';
+import { IUMLElement } from '../../services/uml-element/uml-element';
+import { EditorRepository } from '../../services/editor/editor-repository';
+
+type OwnProps = {};
+
+type StateProps = {
+  readonly: boolean;
+  mode: ApollonMode;
+  elements: UMLElementState;
+  resizingInProgress: boolean;
+  connectingInProgress: boolean;
+  reconnectingInProgress: boolean;
+  hoveringInProgress: boolean;
+  zoomFactor: number;
+};
+
+type DispatchProps = {
+  select: AsyncDispatch<typeof UMLElementRepository.select>;
+  changeSelectionBox: typeof EditorRepository.setSelectionBoxActive;
+};
+
+type Props = OwnProps & StateProps & DispatchProps & CanvasContext;
+
+type LocalState = {
+  selectionStarted: boolean;
+  selectionRectangle: SelectionRectangle;
+};
+
+type SelectionRectangle = {
+  startX: number | undefined;
+  startY: number | undefined;
+  endX: number | undefined;
+  endY: number | undefined;
+};
+
+const enhance = compose<ComponentType<OwnProps>>(
+  withCanvas,
+  connect<StateProps, DispatchProps, OwnProps, ModelState>(
+    (state) => ({
+      readonly: state.editor.readonly,
+      mode: state.editor.mode,
+      elements: state.elements,
+      resizingInProgress: state.resizing.length > 0,
+      connectingInProgress: state.connecting.length > 0,
+      reconnectingInProgress: Object.keys(state.reconnecting).length > 0,
+      hoveringInProgress: state.hovered.length > 0,
+      zoomFactor: state.editor.zoomFactor,
+    }),
+    {
+      select: UMLElementRepository.select,
+      changeSelectionBox: EditorRepository.setSelectionBoxActive,
+    },
+  ),
+);
+
+class MouseEventListenerComponent extends Component<Props, LocalState> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      selectionStarted: false,
+      selectionRectangle: {
+        startX: undefined,
```

Comment:
I would recommend using [`IBoundary`](https://github.com/ls1intum/Apollon/blob/1c2aa378ca8b129b6258b584684ec3b2a6422511/src/main/utils/geometry/boundary.ts#L3-L8) interface, as it would keep the code consistent and allow usage of geometry utilities.